src/main/java/com/example/usermanagementsystemAPIgateway/controller/APIController.java

```diff
@@ -23,17 +25,12 @@
 
         System.out.println(user.getFirstName());
 
-        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl("http://localhost:8081/users")
-                .queryParam("firstName", user.getFirstName())
-                .queryParam("lastName",user.getLastName())
-                .queryParam("emailAddress",user.getEmailAddress());
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON));
+        HttpEntity<UserModel> entity = new HttpEntity<UserModel>(user,headers);
 
         ResponseEntity<UserModel> responseEntity = restTemplate.exchange(
-                builder.build().encode().toUri(),
-                HttpMethod.POST,
-                entity,
-                UserModel.class
-                );
+                "http://localhost:8081/users", HttpMethod.POST, entity, UserModel.class);
```

Comment:
also move `http://localhost:8081` to application property file. It could look like this: `user-managemen-system.url=http://localhost:8081`